### PR TITLE
Add support for NodeJS

### LINKED ISSUE #92: Add support for Node

### PR DESCRIPTION
**Pull request by pior**
## Why

We should support javascript as a language.

Resolves #92

## How

Add a `node` task:
- download the binary distribution from https://nodejs.org/dist/ 
- install the language runtime in the home-data dir (like Go)

### CODE REVIEW

File: pkg/features/node.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	node := helpers.NewNode(cfg, version)
+	if !node.Exists() {
+		return true, nil
+	}
+	env.PrependToPath(node.BinPath())
+	return false, nil
+}
+
+func nodeDeactivate(version string, cfg *config.Config, env *env.Env) {
+	node := helpers.NewNode(cfg, version) // Not great: we use specify an empty version to get the base path with .Path()
```
Comment (mlhamel) [inline pkg/features/node.go]: not sure i understand what is going on here.

File: pkg/helpers/node.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+func (n *Node) Path() string {
+	return n.path
+}
+
+func (n *Node) BinPath() string {
+	return path.Join(n.path, "bin")
+}
+
+func (n *Node) Which(program string) string {
+	return path.Join(n.path, "bin", program)
```
Comment (mlhamel) [inline pkg/helpers/node.go]: were you planning to use `Which` for something else then finding "node". i dont see why getting the `program` as a parameter.

File: pkg/helpers/node.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+func (n *Node) Install() error {
+	if runtime.GOARCH != "amd64" {
+		return fmt.Errorf("Binary distribution are not available for %s architecture", runtime.GOARCH)
```
Comment (mlhamel) [inline pkg/helpers/node.go]: Not sure it's right since most architectures are available. maybe we should rephrase this to something like:
`devbuddy is not currently supporting installation of node on %s architecture` 

?

File: pkg/helpers/node.go (hunk 2 above)
Comment (pior) [reply pkg/helpers/node.go]: It's possible. `npm` is a strong candidate.
